semgrep/semgrep/core_runner.py

```diff
@@ -117,9 +122,10 @@ class CoreRunner:
         This includes properly invoking semgrep-core and parsing the output
     """
 
-    def __init__(self, allow_exec: bool, jobs: int):
+    def __init__(self, allow_exec: bool, jobs: int, output_handler: OutputHandler):
         self._allow_exec = allow_exec
         self._jobs = jobs
+        self._output_handler = output_handler
```

Comment:
the intention is that in a followup diff, this will pave the way to have corerunner emit results as it gets them. I could also imagine some sort of generator interface, but I don't think that giving corerunner a place to report its results is a bad thing

Reply:
I think it will be much cleaner to later modify CoreRunner to expose some streaming interface than have the CoreRunner class need to know about how its results and errors are being used. Much simpler for it to be a layer that interacts with semgrep-core and returns the results of that interaction.

Reply:
I'm not totally sure that's true since there are errors and successes that may both be coming back at different rates, but I've gone that direction for now